teerex/src/lib.rs

```diff
@@ -446,7 +446,13 @@ impl<T: Config> Pallet<T> {
 	}
 
 	fn unregister_silent_workers(now: T::Moment) {
-		let minimum = (now - T::MaxSilenceTime::get()).saturated_into::<u64>();
+		let minimum = now
+			.checked_sub(&T::MaxSilenceTime::get())
+			.unwrap_or_default()
```

Comment:
I think there exists a `saturating_sub` which is  `checked_sub().unwrap_or_default()`, but I am not sure.